x-pack/plugins/fleet/server/routes/setup/handlers.ts

```diff
@@ -21,17 +21,25 @@ export const getFleetStatusHandler: FleetRequestHandler = async (context, reques
       context.core.elasticsearch.client.asInternalUser
     );
 
+    let isReady = true;
     const missingRequirements: GetFleetStatusResponse['missing_requirements'] = [];
+
     if (!isApiKeysEnabled) {
+      isReady = false;
       missingRequirements.push('api_keys');
     }
 
     if (!isFleetServerSetup) {
+      isReady = false;
       missingRequirements.push('fleet_server');
     }
 
+    if (!appContextService.getEncryptedSavedObjectsSetup()?.canEncrypt) {
```

Comment:
I was surprised that this was true for me when I didn't have a key set in my kibana configuration. from the [docs](https://www.elastic.co/guide/en/kibana/current/xpack-security-secure-saved-objects.html):
> If you don’t specify an encryption key, Kibana automatically generates a random key at startup. Every time you restart Kibana, it uses a new ephemeral encryption key and is unable to decrypt saved objects encrypted with another key. To prevent data loss, Kibana might disable features that rely on this encryption until you explicitly set an encryption key.

but from the code it seems like the default is always a fixed value: 
https://github.com/elastic/kibana/blob/80790bc85cdfce861deeb8669bf9f5cf065f98ca/x-pack/plugins/encrypted_saved_objects/server/config.ts#L19

I think it would be good to double check with the Kibana team about this. is there a way to check a user-configured value? from my understanding it doesn't seem like we will ever hit this "encryption key not set" path if a value is always supplied by default

Reply:
It's set by default in dev but not in production (this make testing the feature a little harder as you need to update the code to simulate a missing key) 

The schema is using a conditional here
https://github.com/elastic/kibana/blob/80790bc85cdfce861deeb8669bf9f5cf065f98ca/x-pack/plugins/encrypted_saved_objects/server/config.ts#L16

Reply:
thanks for that tip, do you have reference for it not being set in prod? that seems to conflict with the documentation info

Reply:
Yes it was implemented here https://github.com/elastic/kibana/pull/81511 but looks like the doc say other thing
<img width="779" alt="Screen Shot 2022-04-01 at 4 45 57 PM" src="https://user-images.githubusercontent.com/1336873/161339016-5d70ea57-4c63-4511-9b2a-adf4f9c7375b.png">

Can someone from @elastic/kibana-security can confirm what is the behavior here?

Reply:
I don't have any new information to add here, but it is a bit curious to me that our security on by default feature doesn't seem to auto-populate this.

Reply:
@nchaulet We do not generate an ephemeral encryption key anymore. This was removed in https://github.com/elastic/kibana/pull/81511, so what you're seeing is a docs bug.

> I don't have any new information to add here, but it is a bit curious to me that our security on by default feature doesn't seem to auto-populate this.

@joshdover I agree. We intentionally didn't add this in the first round of work in order to contain scope, and hit our hard deadline of 8.0. We have an issue tracking this:
- https://github.com/elastic/kibana/issues/112259

Reply:
> Yes it was implemented here https://github.com/elastic/kibana/pull/81511 but looks like the doc say other thing

Hmm, it looks look we (I) forgot to update the docs in https://github.com/elastic/kibana/pull/81511, sorry about that.

Since 7.12 we don't generate an ephemeral encryption key in production anymore, the note in documentation is wrong.

> but it is a bit curious to me that our security on by default feature doesn't seem to auto-populate this.

Yeah, we plan to do this in the scope of https://github.com/elastic/kibana/issues/112259, but this work is a blocked at the moment.